extensions/wrapper/wrapper-common-mappers/src/main/java/de/sovity/edc/ext/wrapper/api/common/mappers/dataaddress/http/HttpDataSourceMapper.java

```diff
@@ -0,0 +1,119 @@
+package de.sovity.edc.ext.wrapper.api.common.mappers.dataaddress.http;
+
+import de.sovity.edc.ext.wrapper.api.common.model.UiDataSourceHttpData;
+import de.sovity.edc.ext.wrapper.api.common.model.UiDataSourceOnRequest;
+import de.sovity.edc.utils.jsonld.vocab.Prop;
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonObjectBuilder;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Objects.requireNonNull;
+
+
+@RequiredArgsConstructor
+public class HttpDataSourceMapper {
+    private final HttpHeaderMapper httpHeaderMapper;
+
+    /**
+     * Data Address for type HTTP_DATA
+     *
+     * @param httpData {@link UiDataSourceHttpData}
+     * @return properties for {@link org.eclipse.edc.spi.types.domain.DataAddress}
+     */
+    public Map<String, String> buildDataAddress(@NonNull UiDataSourceHttpData httpData) {
+        var baseUrl = requireNonNull(httpData.getBaseUrl(), "Need baseUrl");
```

Comment:
The error message could be a little more specific
```suggestion
        var baseUrl = requireNonNull(httpData.getBaseUrl(), "baseUrl must not be null");
```